PyDynamic/uncertainty/propagate_filter.py

```diff
@@ -177,17 +176,29 @@ def FIRuncFilter(y, sigma_noise, theta, Utheta=None, shift=0, blow=None, kind="c
         # V has dimension (len(sigma2) + Ntheta) * (len(sigma2) + Ntheta) --> slice a fitting Ulow of dimension (Ntheta x Ntheta)
         UncCov = np.zeros((len(sigma2)))
 
-        for k in range(len(sigma2)):
-            Ulow = V[k:k+Ntheta,k:k+Ntheta]
-            UncCov[k] = np.squeeze(theta.T.dot(Ulow.dot(theta)) + np.abs(np.trace(Ulow.dot(Utheta))))  # static part of uncertainty
+        if isinstance(Utheta, np.ndarray):
```

Comment:
With these `isinstance`-conditions, do you think something could go wrong, if somebody specifies something otherwise valid, which fails these? Alternatively we could check for `is (not) None`, although I cannot come up straight away with a failing example.

Reply:
The docstring requires either an np-array or None as values for `Utheta`, so both checks would lead to the same cases. So I would suggest, to just leave it the way it is at the moment.